osu.Game/Screens/Ranking/ResultsScreen.cs

```diff
@@ -275,6 +277,11 @@ public override bool OnExiting(ScreenExitEvent e)
             if (base.OnExiting(e))
                 return true;
 
+            // This is a stop-gap safety against components holding references to gameplay after exiting the gameplay flow.
+            // Right now, HitEvents are only used up to the results screen. If this changes in the future we need to remove
+            // HitObject references from HitEvent.
+            Score?.HitEvents.Clear();
```

Comment:
Given the rest of the PR, does this do anything meaningful at the moment, or is it just a potential safety?

Reply:
See https://github.com/ppy/osu/pull/26770/commits/eb90ee5415b05dd827464fcd3c715f731da2df1f commit message ;)